### ISSUE TITLE
Automatic Batchsize should be separately calculated for train & val/test

### ISSUE THREAD

**Issue by AAnoosheh** [PyTorchLightning/pytorch-lightning#9219]
## 🚀 Feature

I would like to request auto-batchsize scaling perform separate batchsize operations for train and for val/test.

### Motivation

The auto-batchsize scaling feature is neat, but it also only runs training steps, AFAIK.  
But during val/test, due to the increase in free memory due to lack of gradients, batchsizes can be much larger.

In addition, there are cases I have had where the val/test input sizes are larger than the training ones (i.e. crop images to fit during train, then use whole image during val/test).

### Pitch

Call validate and/or test during the scaling procedure and set the batchsizes directly into the `train_dataloader` and `val/test_dataloader` objects directly, or alternatively as a second class parameter such as `<module>.batch_size_eval`

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

**Comment by tchaton** [comment]
Dear @AAnoosheh,

Would you be interested in contributing such feature ?

Best,
T.C

**Comment by AAnoosheh** [comment]
Hi, I might be able to.

Given my free time, it might be questionable, but if I cannot find a decent workaround for my personal use case, I'll add the feature properly here.
